.github/workflows/publish.yaml

```diff
@@ -0,0 +1,143 @@
+        run: npm install -g konfig-cli
+
+      - name: Detect SDK submodules
+        if: steps.detect-changesets.outputs.has_changesets == 'false'
+        id: sdk-submodules
+        uses: konfig-dev/automation/.github/actions/private/detect-sdk-submodules@main
+        with:
+          konfig_yaml_dir: ${{ inputs.konfig_yaml_dir }}
+
+      - name: Get version bump branch name
+        id: branch-name
+        if: >
+          steps.detect-changesets.outputs.has_changesets == 'false' &&
+          steps.sdk-submodules.outputs.dirs != ''
+        uses: konfig-dev/automation/.github/actions/private/get-branch-name@main
+        with:
+          konfig_yaml_dir: ${{ inputs.konfig_yaml_dir }}
+
+      - name: Set git config
+        if: >
+          steps.detect-changesets.outputs.has_changesets == 'false' &&
+          steps.sdk-submodules.outputs.dirs != ''
+        run: |
+          git config --global user.email "publisher@konfigthis.com"
+          git config --global user.name "konfig-publisher"
+
+      - name: Merge submodule version bump PRs
+        if: >
+          steps.detect-changesets.outputs.has_changesets == 'false' &&
+          steps.sdk-submodules.outputs.dirs != ''
+        working-directory: ${{ inputs.konfig_yaml_dir }}
+        env:
+          KONFIG_API_KEY: ${{ secrets.KONFIG_API_KEY }}
+        run: |
+          git submodule foreach '
+            if [ -n "$(echo "${{ steps.sdk-submodules.outputs.dirs }}" | grep "$displaypath ")" ]; then
+              echo Merging PR in SDK submodule at path: $name
+              # git remote url looks like this: https://github.com/owner/repo.git
+              owner=$(git remote get-url origin | cut -d '/' -f 4)
+              repo=$(git remote get-url origin | cut -d '/' -f 5 | sed 's/\.git.*//')
+              konfig pr-merge \
+                    --owner $owner \
+                    --repo $repo \
+                    --head ${{ steps.branch-name.outputs.branch_name }} \
+                    --base ${{ github.ref_name }}
+            fi
+          '
+
+      - name: Update submodule references
+        if: >
+          steps.detect-changesets.outputs.has_changesets == 'false' &&
+          steps.sdk-submodules.outputs.dirs != ''
+        uses: konfig-dev/automation/.github/actions/private/sync-submodules@main
+        with:
+          konfig_yaml_dir: ${{ inputs.konfig_yaml_dir }}
+          sdk_submodule_dirs: ${{ steps.sdk-submodules.outputs.dirs }}
+          submodules_base_branch: ${{ github.ref_name }}
+          push_to: ${{ github.ref_name }}
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: pre-publish
+    if: needs.pre-publish.outputs.has_changesets == 'false'
+    container: # runs in docker container
+      image: konfigdev/test-and-publish:latest
+    env:
+      NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      PYPI_TOKEN_1: ${{ secrets.PYPI_TOKEN_1 }}
+      PYPI_TOKEN_2: ${{ secrets.PYPI_TOKEN_2 }}
```

Comment:
what if I only have 1 python package (e.g. SnapTrade)?

I have to set `PYPI_TOKEN_1`?

Reply:
yeah.
I can change the name to omit "1" from the first one so that customers with only 1 have a cleaner experience
like 
PYPI_TOKEN
PYPI_TOKEN_2
PYPI_TOKEN_3 
etc.

is that preferable to you?

Reply:
hm, up to you on this one. I could go either way.

Reply:
might just be easier to move on since you've already set it up this way

Reply:
I have a few other changes Im gonna make so Ill just fold it in